### ISSUE TITLE
Why concatenate zeros in the begining of img_logits and txt_logits?

### ISSUE THREAD

**Issue by YoucanBaby** [fudoki-hku/FUDOKI#8]
Why concatenate zeros in the begining of img_logits  and txt_logits?

    img_logits = torch.cat([torch.zeros((img_logits.shape[0], 1, img_logits.shape[2]), device=img_logits.device), img_logits[:, :-1, :]], dim=1)
    txt_logits = torch.cat([torch.zeros((txt_logits.shape[0], 1, txt_logits.shape[2]), device=txt_logits.device), txt_logits[:, :-1, :]], dim=1)

[https://github.com/fudoki-hku/FUDOKI/blob/8bb08e70a3ad6984b6968ba6fb594c29a7241a68/fudoki/janus/models/modeling_vlm.py#L396](url)

**Comment by RainBowLuoCS** [comment]
logits shift used for next token prediction training

**Comment by YoucanBaby** [comment]
Maybe it is related to "attention sink"?

**Comment by jinjinw** [comment]
My apologies for the late reply.

Yes, this is the logits shift operation to inherit the original knowledge of the AR-based model as much as possible. Adding this operation can help the AR-based MLLM with causal attention converge faster to the discrete-flow-based MLLM with bi-directional attention.

**Comment by YoucanBaby** [comment]
> My apologies for the late reply.
> 
> Yes, this is the logits shift operation to inherit the original knowledge of the AR-based model as much as possible. Adding this operation can help the AR-based MLLM with causal attention converge faster to the discrete-flow-based MLLM with bi-directional attention.

ok, I see. Thanks for your reply!
